dlt/extract/pipe.py

```diff
@@ -703,6 +716,12 @@ def stop_background_loop(loop: asyncio.AbstractEventLoop) -> None:
 
         # print("stopping loop")
         if self._async_pool:
+            # wait for all async generators to be closed
+            future = asyncio.run_coroutine_threadsafe(
+                self._async_pool.shutdown_asyncgens(), self._ensure_async_pool()
+            )
+            while not future.done():
+                sleep(self.futures_poll_interval)
```

Comment:
not 100% certain i implemented this right, but i think i did. we need to shut down all the open gens in this pool.

Reply:
@rudolfix I am not sure what you mean with "async". there is a test called `def test_limit_async_resource() -> None:` which tests limiting async resources (which get turned into generators by the wrapper). Can you maybe explain?

Reply:
Do you mean async iterators?